### ISSUE TITLE
Beta7 - addTagHelper syntax clarification

### ISSUE THREAD

**Issue by VaclavElias** [aspnet/Razor#549]
*Why this cannot be done independent from the folder name??*

My **@addTagHelper** didn't work, giving me error *"Cannot resolve TagHelper containing assembly.."*

The reason was because my assembly has got different name from the project folder name.

http://stackoverflow.com/questions/31258452/taghelper-cannot-resolve-taghelper-containing-assembly

I think it doesn't have sense to copy (duplicate) the project to a different folder and also keep on mind that I **must** update **_ViewImports.cshtml** and change the folder name in **@addTagHelper**.

What is the meaning behind this? Thanks :)

**Comment by pranavkm** [comment]
@davidfowl sounds like a DNX issue.

**Comment by davidfowl** [comment]
@VaclavElias is that a runtime error?

**Comment by VaclavElias** [comment]
Whatever it is I don't like it :) See the error below. When I change 

```c#
@addTagHelper "*, Aviik"
```
to 

```c#
@addTagHelper "*, FolderNameOfTheProject"
```
Then all is fine

![addtaghelper](https://cloud.githubusercontent.com/assets/4528464/10119585/e01aa280-6491-11e5-9158-ccacc3f19dba.jpg)

**Comment by dougbu** [comment]
The second part of the `@addTagHelper` syntax is an assembly name in the strict CLR sense of "assembly name". The Razor code does approximately `Assembly.Load(new AssemblyName("Aviik"))` to find the tag helpers of interest. The error above indicates there's no `Aviik.dll` or (equivalently) `Aviik` project in the solution.

So, `Aviik` fails but what does "Aviik" mean in this case? If the project containing tag helpers is in a folder named something else, that will also be the assembly name. Or have you overridden the assembly name in the `project.json` file?

**Comment by VaclavElias** [comment]
A dog replies to my query??? Fair enough :)

Thanks @dougbu I know where the confusion is coming from now after your explanation. In previous version of ASP.NET when I duplicated the project and renamed the project folder and the project name, the Assembly name of the project remained **Aviik**. And this always generated **Aviik.dll**.

In the new World of ASP.NET when I go to project properties, there is not Assembly name but just *Default Namespace* **Aviik** which I wrongly assumed is an assembly. So when I duplicate the project, rename the folder and project name, the *Default Namespace* remains Aviik but automatically the assembly name changes to *Folder Name/Project Name*. And obviously I cannot see anymore **dll** so haven't realized that the new Assembly is the same name as the folder/project name.

I guess this is a bit of lack of my understanding of Assemblies and Namespaces naming that's why I haven't realized that.

Also, I did a test and placed my **TagHelper** to external class library called properly **Aviik** and then all was working as you mentioned as the **Aviik** project was in the solution which generated **Aviik** assembly.

Thanks everyone for your input :)
